src/include/access/tableam.h

```diff
@@ -313,12 +313,14 @@ typedef struct TableAmRoutine
 	 * the scan's behaviour (ScanOptions's SO_ALLOW_*, several may be
 	 * specified, an AM may ignore unsupported ones) and whether the snapshot
 	 * needs to be deallocated at scan_end (ScanOptions's SO_TEMP_SNAPSHOT).
+	 *
+	 * `ctx` is a context pointer that can be used to pass information from analyze or other scan types.
 	 */
 	TableScanDesc (*scan_begin) (Relation rel,
 								 Snapshot snapshot,
 								 int nkeys, struct ScanKeyData *key,
 								 ParallelTableScanDesc pscan,
-								 uint32 flags);
+								 uint32 flags, void *ctx);
```

Comment:
how about add new interface `scan_begin _ctx ` ?

Reply:
Hm...  It is also a good solution. In my opinion, I prefer to add a param to existing interface, for the reason the scan_begin_ctx will always not be used in cbdb kernel. Maybe we can ask @my-ship-it to confirm this.

Reply:
hm, `scan_begin ` add ad param will break other extension that use table-am. like citus etc.

Reply:
Ok, that  is also a problem.